Objects/object.c

```diff
@@ -1728,11 +1743,14 @@ _PyTypes_Init(void)
         return status;
     }
 
+    // XXX We can stop calling _PyObject_SetImmortal() once we change
+    // all the static types to use PyVarObject_HEAD_IMMORTAL_INIT.
 #define INIT_TYPE(TYPE, NAME) \
     do { \
         if (PyType_Ready(TYPE) < 0) { \
             return _PyStatus_ERR("Can't initialize " NAME " type"); \
         } \
+        _PyObject_SetImmortal((PyObject *)TYPE); \
```

Comment:
Don't do that: using static types from multiple interpreters in parallel is unsafe. See for example the PyTypeObject.tp_subclasses issue.

Reply:
Yuck, but yes.  This detail seems to be a particular problem for the lofty idea of sub-interpreters one day eventually having their own GIL...

Reply:
Right.  This is something that has to be addressed before this PR can proceed.